app/services/current-user.ts

```diff
@@ -71,6 +104,14 @@ export default class CurrentUserService extends Service {
             }
         });
     }
+
+    getWaffle(this: CurrentUserService, feature: string) {
+        if (this.get('waffleLoaded')) {
+            return Promise.resolve(this.get('features').isEnabled(feature));
```

Comment:
I like this in general, it's a better interface than I proposed, but I think `Promise.resolve().then(...)` will still be async. `willTransition` will return and the transition will continue a bit before `reload` is called.

It's probably not a big problem in practice, but I worry about having visual cues of a transition before loading the legacy page, and want to minimize the delay to the extent we can. I'm not sure, does that seem like a reasonable worry?